build/package/Microsoft.DotNet.Cli.Installer.PKG.targets

```diff
@@ -1,10 +1,132 @@
+
+        <SdkPkgScriptsDirectory>$(RepoRoot)/packaging/osx/clisdk/scripts</SdkPkgScriptsDirectory>
+        <SdkProductArchiveResourcesDirectory>$(RepoRoot)/packaging/osx/clisdk/resources</SdkProductArchiveResourcesDirectory>
+        
+        <SdkProductArchiveDistributionTemplateFile>$(RepoRoot)/packaging/osx/clisdk/Distribution-Template</SdkProductArchiveDistributionTemplateFile>
+        <SdkProductArchiveDistributionFile>$(PkgIntermediateDirectory)/CLI-SDK-Formatted-Distribution-Template.xml</SdkProductArchiveDistributionFile>
+        
+        <SdkPkgIntermediatePath>$(PkgIntermediateDirectory)/$(SdkComponentId).pkg</SdkPkgIntermediatePath>
+        <SharedFrameworkPkgIntermediatePath>$(PkgIntermediateDirectory)/$(SharedFrameworkComponentId).pkg</SharedFrameworkPkgIntermediatePath>
+        <SharedHostPkgIntermediatePath>$(PkgIntermediateDirectory)/$(SharedHostComponentId).pkg</SharedHostPkgIntermediatePath>
+        <HostFxrPkgIntermediatePath>$(PkgIntermediateDirectory)/$(HostFxrComponentId).pkg</HostFxrPkgIntermediatePath>
+      </PropertyGroup>
+
+      <ItemGroup>
+        <DistributionTemplateReplacement Include="{SharedFxComponentId}">
+          <ReplacementString>$(SharedFrameworkComponentId)</ReplacementString>
+        </DistributionTemplateReplacement>
+        <DistributionTemplateReplacement Include="{SharedHostComponentId}">
+          <ReplacementString>$(SharedHostComponentId)</ReplacementString>
+        </DistributionTemplateReplacement>
+        <DistributionTemplateReplacement Include="{HostFxrComponentId}">
+          <ReplacementString>$(HostFxrComponentId)</ReplacementString>
+        </DistributionTemplateReplacement>
+        <DistributionTemplateReplacement Include="{CLISdkComponentId}">
+          <ReplacementString>$(SdkComponentId)</ReplacementString>
+        </DistributionTemplateReplacement>
+        <DistributionTemplateReplacement Include="{CLISdkNugetVersion}">
+          <ReplacementString>$(SdkVersion)</ReplacementString>
+        </DistributionTemplateReplacement>
+        <DistributionTemplateReplacement Include="{CLISdkBrandName}">
+          <ReplacementString>$(SdkBrandName)</ReplacementString>
+        </DistributionTemplateReplacement>
+        <DistributionTemplateReplacement Include="{SharedFxBrandName}">
+          <ReplacementString>$(SharedFrameworkBrandName)</ReplacementString>
+        </DistributionTemplateReplacement>
+        <DistributionTemplateReplacement Include="{SharedHostBrandName}">
+          <ReplacementString>$(SharedHostBrandName)</ReplacementString>
+        </DistributionTemplateReplacement>
+        <DistributionTemplateReplacement Include="{HostFxrBrandName}">
+          <ReplacementString>$(HostFxrBrandName)</ReplacementString>
+        </DistributionTemplateReplacement>
+      </ItemGroup>
+
+      <!-- Consumed By Publish -->
+      <ItemGroup>
+        <GeneratedInstallers Include="$(SdkInstallerFile);$(CombinedFrameworkSdkHostInstallerFile)" />
```

Comment:
:+1:  This is really nice - package says "Here are my outputs", all publish needs to do is grab these, instead of having to know about all the things that package does.